src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -866,7 +866,7 @@ void AWSClient::AddChecksumToRequest(const std::shared_ptr<Aws::Http::HttpReques
         {
```

Comment:
I'm concerned there is some confusion about checksums and MD5.  MD5 is part of the S3 API (PutObject, for example) and is _not_ a checksum algorithm.  CRC32, CRC64, SHA1, and SHA256 are.

https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html

Reply:
this has some extra info - https://aws.amazon.com/blogs/aws/new-additional-checksum-algorithms-for-amazon-s3/.

Unfortunately, historically how md5 (contentMD5 header) works has not been documented particularly well and there are differences in what scenarios sdks compute it. In general, MD5 is the original checksumming algo that s3 has (and its used purely for integrity purposes and s3 does not rely on any security properties of md5) and other algoes, like crc32/sha1/etc... were added as an alternative to MD5 - meaning if they are set, md5 is not computed anymore.

Reply:
Please consider the following before making changes in this area: other cloud object storage systems like GCS, Alibaba, Seagate, Oracle; and on-premises systems like IBM, Dell Isilon all implement the S3 API and still require the use of Content-MD5.  If S3 does not want to use MD5 any more, fine, but removing Content-MD5 from the SDK breaks the whole ecosystem.